packages/ngx-qgrid-plugins/src/lib/data-provider/data-provider.component.ts

```diff
@@ -0,0 +1,46 @@
+import { HttpClient } from '@angular/common/http';
+import {ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, OnInit, Output} from '@angular/core';
+import { GridPlugin, StateAccessor } from '@qgrid/ngx';
+import { PipeUnit } from '@qgrid/core/public-api';
+
+@Component({
+	selector: 'q-data-provider',
+	template: '',
+	providers: [GridPlugin, StateAccessor],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DataProviderComponent implements OnInit, OnChanges {
+	nextReference: (args) => void;
+
+	@Input('data') set tempData(value) {
```

Comment:
just rename it to `data` please not `tempData` and value should have time `any[]`

Reply:
and please make like following

```typescript
const next = this.next;
if (next) {
   this.next = null; 
   next(value);
}
```